src/apply_manual_codes.py

```diff
@@ -0,0 +1,168 @@
+                        "CodeID"] != \
+                                            plan.code_scheme.get_code_with_control_code(
+                                                Codes.NOT_REVIEWED).code_id
+                    
+                    if binary_label_present and not reasons_label_present:
+                        if binary_code.code_type == "Control":
+                            control_code = binary_code.control_code
+                            reasons_code = plan.code_scheme.get_code_with_control_code(control_code)
+
+                            reasons_label = CleaningUtils.make_label_from_cleaner_code(
+                                plan.code_scheme, reasons_code,
+                                Metadata.get_call_location(), origin_name="Pipeline Code Synchronisation")
+
+                            td.append_data(
+                                {plan.coded_field: [reasons_label.to_dict()]},
+                                Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string())
+                            )
+                        else:
+                            assert binary_code.code_type == "Normal"
+
+                            nc_label = CleaningUtils.make_label_from_cleaner_code(
+                                plan.code_scheme, plan.code_scheme.get_code_with_control_code(Codes.NOT_CODED),
+                                Metadata.get_call_location(), origin_name="Pipeline Code Synchronisation"
+                            )
+                            td.append_data(
+                                {plan.coded_field: [nc_label.to_dict()]},
+                                Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string())
+                            )
+        
+        # Merge manually coded demog and survey files into the cleaned dataset
+        # Recursion depth currently exceeding
+        # TODO: Investigate/address the cause of this.
+        sys.setrecursionlimit(10000)
+        for plan in PipelineConfiguration.DEMOGS_CODING_PLANS:
+            f = None
+            try:
+                coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable(
+                    user, data, plan.id_field, {plan.coded_field: plan.code_scheme}, f)
+            finally:
+                if f is not None:
+                    f.close()
+        
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
+            f = None
+            try:
+                coda_input_path = path.join(coda_input_dir, plan.coda_filename)
+                if path.exists(coda_input_path):
+                    f = open(coda_input_path, "r")
+                TracedDataCodaV2IO.import_coda_2_to_traced_data_iterable(
```

Comment:
Actually shouldn't you be treating this like a radio question? Given that your follow-up surveys need multi-coding shouldn't this be using the multi-coded importer?

Reply:
Updated to Multi coded importer thanks